diff --git a/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h b/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h
index 963c1d8e..73487edd 100644
--- a/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h
+++ b/storage/rocksdb/rocksdb/db/compaction/compaction_iteration_stats.h
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include "rocksdb/rocksdb_namespace.h"
 
 struct CompactionIterationStats {
diff --git a/storage/rocksdb/rocksdb/env/composite_env_wrapper.h b/storage/rocksdb/rocksdb/env/composite_env_wrapper.h
index fbc0b93f..e10d617b 100644
--- a/storage/rocksdb/rocksdb/env/composite_env_wrapper.h
+++ b/storage/rocksdb/rocksdb/env/composite_env_wrapper.h
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include "rocksdb/env.h"
 #include "rocksdb/file_system.h"
 
diff --git a/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h b/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h
index c7f93b4c..3c2ab805 100644
--- a/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h
+++ b/storage/rocksdb/rocksdb/include/rocksdb/utilities/checkpoint.h
@@ -8,6 +8,7 @@
 #pragma once
 #ifndef ROCKSDB_LITE
 
+#include <cstdint>
 #include <string>
 #include <vector>
 #include "rocksdb/status.h"
diff --git a/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h b/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
index f356395f..6eaf7173 100644
--- a/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
+++ b/storage/rocksdb/rocksdb/table/block_based/data_block_hash_index.h
@@ -7,6 +7,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "rocksdb/slice.h"
 
diff --git a/storage/rocksdb/rocksdb/util/string_util.h b/storage/rocksdb/rocksdb/util/string_util.h
index a761be66..2e57eda2 100644
--- a/storage/rocksdb/rocksdb/util/string_util.h
+++ b/storage/rocksdb/rocksdb/util/string_util.h
@@ -10,6 +10,7 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <cstdint>
 
 #include "rocksdb/rocksdb_namespace.h"
 
diff --git a/tpool/aio_linux.cc b/tpool/aio_linux.cc
index 5d01c588..90888622 100644
--- a/tpool/aio_linux.cc
+++ b/tpool/aio_linux.cc
@@ -20,6 +20,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02111 - 1301 USA*/
 # include <atomic>
 # include <libaio.h>
 # include <sys/syscall.h>
+# include <cstdio>
 
 /**
   Invoke the io_getevents() system call, without timeout parameter.

