diff -Nur dhcpcd-7.0.0.orig/src/dhcpcd.conf dhcpcd-7.0.0/src/dhcpcd.conf
--- dhcpcd-7.0.0.orig/src/dhcpcd.conf	2018-01-01 05:17:56.000000000 -0600
+++ dhcpcd-7.0.0/src/dhcpcd.conf	2018-01-02 04:19:53.997148193 -0600
@@ -30,7 +30,7 @@
 option interface_mtu
 
 # Most distributions have NTP support.
-#option ntp_servers
+option ntp_servers
 
 # A ServerID is required by RFC2131.
 require dhcp_server_identifier
