diff -Nur dhcpcd-7.0.0.orig/src/dhcpcd.conf dhcpcd-7.0.0/src/dhcpcd.conf
--- dhcpcd-7.0.0.orig/src/dhcpcd.conf	2018-01-01 05:17:56.000000000 -0600
+++ dhcpcd-7.0.0/src/dhcpcd.conf	2018-01-02 04:15:43.026058268 -0600
@@ -39,3 +39,7 @@
 #slaac hwaddr
 # OR generate Stable Private IPv6 Addresses based from the DUID
 slaac private
+
+# We run wpa_supplicant from rc.inet1 and/or NM handles it on its own
+nohook wpa_supplicant
+
