diff -aurN reiserfsprogs-3.6.27/include/misc.h reiserfsprogs-3.6.27-mod/include/misc.h
--- reiserfsprogs-3.6.27/include/misc.h	2017-07-25 01:34:11.000000000 -0000
+++ reiserfsprogs-3.6.27-mod/include/misc.h	2018-08-08 15:07:06.101179797 -0000
@@ -73,7 +73,7 @@
 int get_dma_info(dma_info_t *dma_info);
 void clean_after_dma_check(int fd, dma_info_t *dma_info);
 
-int valid_offset(int fd, loff_t offset);
+int valid_offset(int fd, off_t offset);
 unsigned long count_blocks(const char *filename, int blocksize);
 
 void print_how_far(FILE * fp, unsigned long *passed, unsigned long total,
diff -aurN reiserfsprogs-3.6.27/include/reiserfs_lib.h reiserfsprogs-3.6.27-mod/include/reiserfs_lib.h
--- reiserfsprogs-3.6.27/include/reiserfs_lib.h	2016-06-07 14:22:56.000000000 -0000
+++ reiserfsprogs-3.6.27-mod/include/reiserfs_lib.h	2018-08-08 15:06:30.796179932 -0000
@@ -297,8 +297,8 @@
 char *key_of_what(const struct reiserfs_key *key);
 int type_unknown(const struct reiserfs_key *key);
 void set_type(int format, struct reiserfs_key *key, int type);
-void set_offset(int format, struct reiserfs_key *key, loff_t offset);
-void set_type_and_offset(int format, struct reiserfs_key *key, loff_t offset,
+void set_offset(int format, struct reiserfs_key *key, off_t offset);
+void set_type_and_offset(int format, struct reiserfs_key *key, off_t offset,
 			 int type);
 
 typedef int (*check_unfm_func_t) (reiserfs_filsys_t , __u32);
