diff -aurN digikam-7.6.0/core/utilities/extrasupport/addressbook/akonadiiface.cpp digikam-7.6.0-mod/core/utilities/extrasupport/addressbook/akonadiiface.cpp
--- digikam-7.6.0/core/utilities/extrasupport/addressbook/akonadiiface.cpp	2022-03-01 09:42:07.000000000 +0100
+++ digikam-7.6.0-mod/core/utilities/extrasupport/addressbook/akonadiiface.cpp	2022-04-23 18:33:43.340303184 +0200
@@ -49,7 +49,11 @@
 #include <AkonadiCore/Item>
 #endif
 
+#if AKONADI_VERSION >= QT_VERSION_CHECK(5, 19, 80)
+#include <Akonadi/ContactSearchJob>
+#else
 #include <Akonadi/Contact/ContactSearchJob>
+#endif
 #include <KContacts/Addressee>
 
 #if defined(Q_OS_DARWIN) && defined(Q_CC_CLANG)
