diff -aurN jfsutils-1.1.15/libfs/devices.c jfsutils-1.1.15-mod/libfs/devices.c
--- jfsutils-1.1.15/libfs/devices.c	2009-03-23 16:37:05.000000000 +0100
+++ jfsutils-1.1.15-mod/libfs/devices.c	2018-08-05 19:45:32.106198848 +0200
@@ -40,6 +40,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
+#include <sys/sysmacros.h>
 #if defined(__DragonFly__)
 #include <machine/param.h>
 #include <sys/diskslice.h>
