diff -aurN kbd-1.15.3/src/getkeycodes.c kbd-1.15.3-mod/src/getkeycodes.c
--- kbd-1.15.3/src/getkeycodes.c	2011-04-26 22:29:35.000000000 +0200
+++ kbd-1.15.3-mod/src/getkeycodes.c	2020-07-28 14:43:56.014098881 +0200
@@ -24,7 +24,7 @@
 	int fd;
 	unsigned int sc, sc0;
 	struct kbkeycode a;
-	int old_kernel = 0;
+/*	int old_kernel = 0; */
 
 	set_progname(argv[0]);
 
@@ -43,7 +43,7 @@
 	a.scancode = 0;
 	a.keycode = 0;
 	if (ioctl(fd, KDGETKEYCODE, &a)) {
-		old_kernel = 1;
+/*		old_kernel = 1; */
 		sc0 = 89;
 	} else
 	for (sc0 = 1; sc0 <= 88; sc0++) {
