diff -ruBN /tmp/aaa_terminfo/aaa_terminfo.SlackBuild /source/a/aaa_terminfo/aaa_terminfo.SlackBuild
--- /tmp/aaa_terminfo/aaa_terminfo.SlackBuild	2018-11-13 22:15:24.000000000 +0000
+++ /source/a/aaa_terminfo/aaa_terminfo.SlackBuild	2021-01-06 11:55:37.000000000 +0000
@@ -26,7 +26,7 @@
 
 PKGNAM=aaa_terminfo
 # Note the version of ncurses in use:
-VERSION=${VERSION:-$(PRINT_PACKAGE_NAME=yes ./ncurses.SlackBuild | cut -f 2 -d -)}
+VERSION=${VERSION:-$(PRINT_PACKAGE_NAME=yes sh ./ncurses.SlackBuild | cut -f 2 -d -)}
 BUILD=${BUILD:-1}
 TMP=${TMP:-/tmp}
 PKG=$TMP/package-${PKGNAM}
@@ -50,14 +50,14 @@
 fi
 
 if [ "$ARCH" = "x86_64" ]; then
-  LIBDIRSUFFIX="64"
+  LIBDIRSUFFIX=""
 else
   LIBDIRSUFFIX=""
 fi
 
 # If we don't have an ncurses package available in $TMP, we need to build it:
-if ! /bin/ls $TMP/$(PRINT_PACKAGE_NAME=yes $CWD/ncurses.SlackBuild) 1> /dev/null 2> /dev/null ; then
-  TMP=$TMP $CWD/ncurses.SlackBuild
+if ! /bin/ls $TMP/$(PRINT_PACKAGE_NAME=yes sh $CWD/ncurses.SlackBuild) 1> /dev/null 2> /dev/null ; then
+  TMP=$TMP sh $CWD/ncurses.SlackBuild
 fi
 
 rm -rf $PKG
@@ -66,7 +66,7 @@
 cd $PKG
 mkdir cruft
 cd cruft
-explodepkg $TMP/$(PRINT_PACKAGE_NAME=yes $CWD/ncurses.SlackBuild)
+explodepkg $TMP/$(PRINT_PACKAGE_NAME=yes sh $CWD/ncurses.SlackBuild)
 sh install/doinst.sh
 cd ..

