diff -aurN xdm-20220430_0f46b67/xdm/genauth.c xdm-20220430_0f46b67-mod/xdm/genauth.c
--- xdm-20220430_0f46b67/xdm/genauth.c	2022-08-14 10:05:53.000000000 +0200
+++ xdm-20220430_0f46b67-mod/xdm/genauth.c	2022-08-14 10:19:34.519010338 +0200
@@ -40,14 +40,6 @@
 
 #include <errno.h>
 
-#ifdef HAVE_ARC4RANDOM
-# ifdef __linux__
-#  include <bsd/stdlib.h>
-# else
-#  include <stdlib.h>
-# endif
-#endif
-
 #include <time.h>
 #define Time_t time_t
 
