diff -aurN vsftpd-3.0.3/sysstr.c vsftpd-3.0.3-mod/sysstr.c
--- vsftpd-3.0.3-mod/sysstr.c	2020-10-30 17:57:11.051907463 -0000
+++ vsftpd-3.0.3/sysstr.c	2011-12-14 10:26:39.000000000 -0000
@@ -74,7 +74,7 @@
 int
 str_open(const struct mystr* p_str, const enum EVSFSysStrOpenMode mode)
 {
-  enum EVSFSysUtilOpenMode open_mode = kVSFSysStrOpenUnknown;
+  enum EVSFSysUtilOpenMode open_mode = (enum EVSFSysUtilOpenMode) kVSFSysStrOpenUnknown;
   switch (mode)
   {
     case kVSFSysStrOpenReadOnly:
