diff -aurN kde-baseapps-4.14.3/konq-plugins/validators/tidy_validator.cpp kde-baseapps-4.14.3-mod/konq-plugins/validators/tidy_validator.cpp
--- kde-baseapps-4.14.3/konq-plugins/validators/tidy_validator.cpp	2014-10-30 08:15:16.000000000 +0100
+++ kde-baseapps-4.14.3-mod/konq-plugins/validators/tidy_validator.cpp	2020-10-10 11:55:25.788888494 +0200
@@ -25,7 +25,7 @@
 
 #include <kdebug.h>
 
-#include <buffio.h>
+#include <tidybuffio.h>
 #include <tidy.h>
 
 #include <config-konq-validator.h>
