diff -aurN samba-4.16.4/lib/replace/system/filesys.h samba-4.16.4-mod/lib/replace/system/filesys.h
--- samba-4.16.4/lib/replace/system/filesys.h	2022-01-24 11:26:58.905306300 +0100
+++ samba-4.16.4-mod/lib/replace/system/filesys.h	2022-08-11 09:14:05.976876505 +0200
@@ -36,10 +36,6 @@
 #include <sys/param.h>
 #endif
 
-#ifdef HAVE_SYS_MOUNT_H
-#include <sys/mount.h>
-#endif
-
 #ifdef HAVE_MNTENT_H
 #include <mntent.h>
 #endif
