diff -ruBN /tmp/etc/doinst.sh /source/a/etc/doinst.sh
--- /tmp/etc/doinst.sh	1970-01-01 00:00:00.000000000 +0000
+++ /source/a/etc/doinst.sh	2021-01-03 11:27:06.000000000 +0000
@@ -0,0 +1,113 @@
+#!/bin/sh
+config() {
+  NEW="$1"
+  OLD="`dirname $NEW`/`basename $NEW .new`"
+  # If there's no config file by that name, mv it over:
+  if [ ! -r $OLD ]; then
+    mv $NEW $OLD
+  elif [ "`cat $OLD | md5sum`" = "`cat $NEW | md5sum`" ]; then # toss the redundant copy
+    rm $NEW
+  fi
+  # Otherwise, we leave the .new copy for the admin to consider...
+}
+
+# First, make sure any new entries in passwd/shadow/group are added:
+if [ -r etc/passwd -a -r etc/passwd.new ]; then
+  cat etc/passwd.new | while read line ; do
+    if ! grep -q "^$(echo $line | cut -f 1 -d :):" etc/passwd ; then
+      echo $line >> etc/passwd
+    fi
+  done
+fi
+if [ -r etc/shadow -a -r etc/shadow.new ]; then
+  cat etc/shadow.new | while read line ; do
+    if ! grep -q "^$(echo $line | cut -f 1 -d :):" etc/shadow ; then
+      echo $line >> etc/shadow
+    fi
+  done
+fi
+if [ -r etc/group -a -r etc/group.new ]; then
+  cat etc/group.new | while read line ; do
+    if ! grep -q "^$(echo $line | cut -f 1 -d :):" etc/group ; then
+      echo $line >> etc/group
+    fi
+  done
+fi
+
+config etc/mtab.new
+config etc/motd.new
+config etc/group.new
+config etc/csh.login.new
+config etc/profile.new
+config etc/hosts.new
+config etc/inputrc.new
+config etc/shadow.new
+config etc/passwd.new
+config etc/printcap.new
+config etc/networks.new
+config etc/HOSTNAME.new
+config etc/gshadow.new
+config etc/issue.new
+config etc/securetty.new
+config etc/shells.new
+config etc/services.new
+config etc/issue.net.new
+config etc/nsswitch.conf.new
+config etc/profile.d/lang.csh.new
+config etc/profile.d/lang.sh.new
+config etc/profile.d/z-dot-in-non-root-path.csh.new
+config etc/profile.d/z-dot-in-non-root-path.sh.new
+config var/log/lastlog.new
+config var/log/wtmp.new
+config var/run/utmp.new
+
+if [ -r etc/ld.so.conf.new -a -r etc/ld.so.conf ]; then
+  # Ensure that ld.so.conf contains the minimal set of paths:
+  # (eliminate ld.so.conf.d line when adding paths to avoid repeats)
+  cat etc/ld.so.conf | grep -v ld.so.conf.d | while read pathline ; do
+    if ! grep "^${pathline}$" etc/ld.so.conf.new 1> /dev/null 2> /dev/null ; then
+      echo "$pathline" >> etc/ld.so.conf.new
+    fi
+  done
+  cp etc/ld.so.conf.new etc/ld.so.conf
+fi
+
+# Clean up useless non-examples:
+rm -f etc/mtab.new
+rm -f etc/motd.new
+rm -f etc/hosts.new
+#rm -f etc/shadow.new
+rm -f etc/networks.new
+rm -f etc/HOSTNAME.new
+#rm -f etc/gshadow.new
+rm -f etc/shells.new
+rm -f etc/printcap.new
+#rm -f etc/issue.new
+rm -f etc/issue.net.new
+#rm -f etc/profile.d/lang.csh.new
+#rm -f etc/profile.d/lang.sh.new
+rm -f var/run/utmp.new
+rm -f var/log/lastlog.new
+rm -f var/log/wtmp.new
+
+# Make sure $HOME is correct for user sddm:
+chroot . /usr/sbin/usermod -d /var/lib/sddm sddm > /dev/null 2> /dev/null
+# Make sure that sddm is a member of group video:
+chroot . /usr/sbin/usermod --groups video sddm > /dev/null 2> /dev/null
+
+# Also ensure ownerships/perms:
+chown root.utmp var/run/utmp var/log/wtmp
+chmod 664 var/run/utmp var/log/wtmp
+chown root.shadow etc/shadow etc/gshadow
+chmod 640 etc/shadow etc/gshadow
+
+# Match permissions on any leftover config z-dot-in-non-root-path scripts
+# to prevent anyone who turned them on from accidentally losing that setting
+# by moving the .new script into place:
+if [ -r etc/profile.d/z-dot-in-non-root-path.csh.new ]; then
+  touch -r etc/profile.d/z-dot-in-non-root-path.csh etc/profile.d/z-dot-in-non-root-path.csh.new
+fi
+if [ -r etc/profile.d/z-dot-in-non-root-path.sh.new ]; then
+  touch -r etc/profile.d/z-dot-in-non-root-path.sh etc/profile.d/z-dot-in-non-root-path.sh.new
+fi
+
diff -ruBN /tmp/etc/etc.SlackBuild /source/a/etc/etc.SlackBuild
--- /tmp/etc/etc.SlackBuild	2020-05-15 03:07:01.000000000 +0000
+++ /source/a/etc/etc.SlackBuild	2021-01-03 11:27:06.000000000 +0000
@@ -53,12 +53,21 @@
 cd $PKG
 explodepkg $CWD/_etc.tar.gz
 # Fix for ARCH:
-sed -i -e "s#/i586-slackware-linux#/${ARCH}-slackware-linux#" etc/ld.so.conf.new
+echo "/lib
+/usr/lib
+/usr/local/lib
+/usr/$ARCH-slackware-linux-musl/lib" > $PKG/etc/ld-musl-$ARCH.path.sample
+[ $ARCH = x86_64 ] && echo "/lib64
+/usr/lib64
+/usr/local/lib64
+/usr/$ARCH-slackware-linux-musl/lib64" >> $PKG/etc/ld-musl-$ARCH.path.sample
+true
 # Fix paths if lib64 is needed:
 if [ "$ARCH" = "x86_64" ]; then
-  sed -i -e "s#/lib#/lib64#" etc/ld.so.conf.new
+true
 fi
 zcat $CWD/nsswitch.conf.gz > $PKG/etc/nsswitch.conf.new
+rm -rf etc/ld.so.conf*
 
 # Check to make sure that shadow has the same number of entries
 # as passwd:
@@ -75,7 +84,7 @@
 
 mkdir -p $PKG/install
 cat $CWD/slack-desc > $PKG/install/slack-desc
-zcat $CWD/doinst.sh.gz > $PKG/install/doinst.sh
+cat $CWD/doinst.sh > $PKG/install/doinst.sh
 
 # Build the package:
 cd $PKG
diff -ruBN /tmp/etc/etc.SlackBuild.log /source/a/etc/etc.SlackBuild.log
--- /tmp/etc/etc.SlackBuild.log	1970-01-01 00:00:00.000000000 +0000
+++ /source/a/etc/etc.SlackBuild.log	2021-01-03 11:27:06.000000000 +0000
@@ -0,0 +1,102 @@
+Exploding package /source/a/etc/_etc.tar.gz in current directory:
+./
+etc/
+etc/HOSTNAME.new
+etc/csh.login.new
+etc/group.new
+etc/gshadow.new
+etc/hosts.new
+etc/inputrc.new
+etc/issue.net.new
+etc/issue.new
+etc/ld.so.conf.d/
+etc/ld.so.conf.new
+etc/motd.new
+etc/mtab.new
+etc/networks.new
+etc/nsswitch.conf.new
+etc/passwd.new
+etc/printcap.new
+etc/profile.d/
+etc/profile.d/lang.csh.new
+etc/profile.d/lang.sh.new
+etc/profile.d/z-dot-in-non-root-path.csh.new
+etc/profile.d/z-dot-in-non-root-path.sh.new
+etc/profile.new
+etc/securetty.new
+etc/services.new
+etc/shadow.new
+etc/shells.new
+etc/skel/
+usr/
+var/
+var/log/
+var/log/lastlog.new
+var/log/wtmp.new
+var/run/
+var/run/utmp.new
+
+Slackware package maker, version 3.14159265.
+
+Searching for symbolic links:
+
+No symbolic links were found, so we won't make an installation script.
+You can make your own later in ./install/doinst.sh and rebuild the
+package if you like.
+
+This next step is optional - you can set the directories in your package
+to some sane permissions. If any of the directories in your package have
+special permissions, then DO NOT reset them here!
+
+Would you like to reset all directory permissions to 755 (drwxr-xr-x) and
+directory ownerships to root.root ([y]es, [n]o)? n
+
+Creating Slackware package:  /tmp/etc-15.0-x86_64-11.txz
+
+./
+etc/
+etc/HOSTNAME.new
+etc/csh.login.new
+etc/group.new
+etc/gshadow.new
+etc/hosts.new
+etc/inputrc.new
+etc/issue.net.new
+etc/issue.new
+etc/ld-musl-x86_64.path.sample
+etc/motd.new
+etc/mtab.new
+etc/networks.new
+etc/nsswitch.conf.new
+etc/passwd.new
+etc/printcap.new
+etc/profile.d/
+etc/profile.d/lang.csh.new
+etc/profile.d/lang.sh.new
+etc/profile.d/z-dot-in-non-root-path.csh.new
+etc/profile.d/z-dot-in-non-root-path.sh.new
+etc/profile.new
+etc/securetty.new
+etc/services.new
+etc/shadow.new
+etc/shells.new
+etc/skel/
+install/
+install/doinst.sh
+install/slack-desc
+usr/
+var/
+var/log/
+var/log/lastlog.new
+var/log/wtmp.new
+var/run/
+var/run/utmp.new
+WARNING:  zero length file etc/mtab.new
+WARNING:  zero length file etc/gshadow.new
+WARNING:  zero length file etc/issue.net.new
+WARNING:  zero length file var/run/utmp.new
+WARNING:  zero length file var/log/wtmp.new
+WARNING:  zero length file var/log/lastlog.new
+
+Slackware package /tmp/etc-15.0-x86_64-11.txz created.
+
