diff -aurN qt-everywhere-src-5.15.0.old/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp qt-everywhere-src-5.15.0/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp
--- qt-everywhere-src-5.15.0.old/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp	2020-08-13 09:34:45.769410760 +0200
+++ qt-everywhere-src-5.15.0/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp	2020-08-13 09:35:21.985781973 +0200
@@ -43,6 +43,7 @@
 #include "qimage.h"
 #include "qvariant.h"
 #include "qcolor.h"
+#include "cmath"
 
 #include <jasper/jasper.h>
 
