diff -Nrbu seamonkey-2.49.4/seamonkey-2.49.4/mozilla/tools/profiler/core/platform.h seamonkey-2.49.4-OK/seamonkey-2.49.4/mozilla/tools/profiler/core/platform.h
--- seamonkey-2.49.4/seamonkey-2.49.4/mozilla/tools/profiler/core/platform.h	2018-07-12 05:19:15.000000000 +0300
+++ seamonkey-2.49.4-OK/seamonkey-2.49.4/mozilla/tools/profiler/core/platform.h	2019-07-27 20:31:47.675086876 +0300
@@ -65,10 +65,7 @@
 #include <unistd.h>
 #if !defined(__BIONIC__)
 #include <sys/syscall.h>
-static inline pid_t gettid()
-{
-  return (pid_t) syscall(SYS_gettid);
-}
+
 #endif
 #endif
 
