diff -aurN pipewire-0.3.50/spa/include/spa/utils/string.h pipewire-0.3.50-mod/spa/include/spa/utils/string.h
--- pipewire-0.3.50/spa/include/spa/utils/string.h	2022-04-13 12:41:34.000000000 +0200
+++ pipewire-0.3.50-mod/spa/include/spa/utils/string.h	2022-04-17 17:04:18.214443582 +0200
@@ -276,10 +276,11 @@
 static inline float spa_strtof(const char *str, char **endptr)
 {
 	static locale_t locale = NULL;
+	locale_t prev;
 	float v;
 	if (SPA_UNLIKELY(locale == NULL))
 		locale = newlocale(LC_ALL_MASK, "C", NULL);
-	locale_t prev = uselocale(locale);
+	prev = uselocale(locale);
 	v = strtof(str, endptr);
 	uselocale(prev);
 	return v;
@@ -319,10 +320,11 @@
 static inline double spa_strtod(const char *str, char **endptr)
 {
 	static locale_t locale = NULL;
+	locale_t prev;
 	double v;
 	if (SPA_UNLIKELY(locale == NULL))
 		locale = newlocale(LC_ALL_MASK, "C", NULL);
-	locale_t prev = uselocale(locale);
+	prev = uselocale(locale);
 	v = strtod(str, endptr);
 	uselocale(prev);
 	return v;
