diff -aurN autofs-5.1.8/modules/parse_amd.c autofs-5.1.8-mod/modules/parse_amd.c
--- autofs-5.1.8/modules/parse_amd.c	2021-10-19 04:56:28.000000000 +0200
+++ autofs-5.1.8-mod/modules/parse_amd.c	2022-08-12 04:16:47.092856865 +0200
@@ -27,7 +27,6 @@
 #include <sys/utsname.h>
 #include <netinet/in.h>
 #include <sys/mount.h>
-#include <linux/fs.h>
 
 #define MODULE_PARSE
 #include "automount.h"
diff -aurN autofs-5.1.8/modules/parse_sun.c autofs-5.1.8-mod/modules/parse_sun.c
--- autofs-5.1.8/modules/parse_sun.c	2021-10-19 04:56:28.000000000 +0200
+++ autofs-5.1.8-mod/modules/parse_sun.c	2022-08-12 04:13:28.733078282 +0200
@@ -30,7 +30,6 @@
 #include <sys/utsname.h>
 #include <netinet/in.h>
 #include <sys/mount.h>
-#include <linux/fs.h>
 
 #define MODULE_PARSE
 #include "automount.h"
