diff -aurN squashfs4.3/squashfs-tools/mksquashfs.c squashfs4.3-mod/squashfs-tools/mksquashfs.c
--- squashfs4.3/squashfs-tools/mksquashfs.c	2018-08-08 16:16:48.776163841 -0000
+++ squashfs4.3-mod/squashfs-tools/mksquashfs.c	2018-08-08 16:27:20.574161431 -0000
@@ -44,6 +44,7 @@
 #include <setjmp.h>
 #include <sys/types.h>
 #include <sys/mman.h>
+#include <sys/sysmacros.h>
 #include <pthread.h>
 #include <regex.h>
 #include <fnmatch.h>
diff -aurN squashfs4.3/squashfs-tools/unsquashfs.c squashfs4.3-mod/squashfs-tools/unsquashfs.c
--- squashfs4.3/squashfs-tools/unsquashfs.c	2018-08-08 16:16:48.680163842 -0000
+++ squashfs4.3-mod/squashfs-tools/unsquashfs.c	2018-08-08 16:35:00.492159677 -0000
@@ -35,6 +35,7 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
+#include <sys/sysmacros.h>
 #include <limits.h>
 #include <ctype.h>
 
