diff -aurN fdutils-5.5/src/diskseekd.c fdutils-5.5-mod/src/diskseekd.c
--- fdutils-5.5/src/diskseekd.c	2002-11-02 14:36:42.000000000 +0100
+++ fdutils-5.5-mod/src/diskseekd.c	2018-08-05 19:33:51.424201521 +0200
@@ -14,6 +14,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <linux/major.h>
+#include <sys/sysmacros.h>
 #include "enh_options.h"
 #include <sys/signal.h>
 #include <errno.h>
diff -aurN fdutils-5.5/src/driveprm.c fdutils-5.5-mod/src/driveprm.c
--- fdutils-5.5/src/driveprm.c	2005-03-03 21:45:38.000000000 +0100
+++ fdutils-5.5-mod/src/driveprm.c	2018-08-05 19:32:15.672201886 +0200
@@ -8,6 +8,7 @@
 #include <linux/fd.h>
 #include <sys/ioctl.h>
 #include <linux/major.h>
+#include <sys/sysmacros.h>
 #include "parse.h"
 #include "driveprm.h"
 #include "driveprmP.h"
diff -aurN fdutils-5.5/src/floppymeter.c fdutils-5.5-mod/src/floppymeter.c
--- fdutils-5.5/src/floppymeter.c	2002-11-02 14:37:12.000000000 +0100
+++ fdutils-5.5-mod/src/floppymeter.c	2018-08-05 19:34:17.650201421 +0200
@@ -14,6 +14,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <linux/major.h>
+#include <sys/sysmacros.h>
 #include <getopt.h>
 #include "enh_options.h"
 #include "fdutils.h"
diff -aurN fdutils-5.5/src/superformat.c fdutils-5.5-mod/src/superformat.c
--- fdutils-5.5/src/superformat.c	2018-08-05 19:35:01.360201254 +0200
+++ fdutils-5.5-mod/src/superformat.c	2018-08-05 19:32:51.827201748 +0200
@@ -47,6 +47,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <linux/major.h>
+#include <sys/sysmacros.h>
 #include <errno.h>
 #include "enh_options.h"
 #include "mediaprm.h"
diff -aurN fdutils-5.5/src/xdfcopy.c fdutils-5.5-mod/src/xdfcopy.c
--- fdutils-5.5/src/xdfcopy.c	2003-07-18 08:27:48.000000000 +0200
+++ fdutils-5.5-mod/src/xdfcopy.c	2018-08-05 19:33:26.180201617 +0200
@@ -18,6 +18,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <linux/major.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <errno.h>
 #include "fdutils.h"
