diff -aurN postfix-3.4.6/src/dns/dns_str_resflags.c postfix-3.4.6.new/src/dns/dns_str_resflags.c
--- postfix-3.4.6/src/dns/dns_str_resflags.c	2016-05-15 17:20:14.000000000 +0200
+++ postfix-3.4.6.new/src/dns/dns_str_resflags.c	2019-08-03 16:08:59.364154893 +0200
@@ -52,7 +52,7 @@
 static const LONG_NAME_MASK resflag_table[] = {
     "RES_INIT", RES_INIT,
     "RES_DEBUG", RES_DEBUG,
-    "RES_AAONLY", RES_AAONLY,
+/*    "RES_AAONLY", RES_AAONLY, */
     "RES_USEVC", RES_USEVC,
     "RES_PRIMARY", RES_PRIMARY,
     "RES_IGNTC", RES_IGNTC,
@@ -60,10 +60,10 @@
     "RES_DEFNAMES", RES_DEFNAMES,
     "RES_STAYOPEN", RES_STAYOPEN,
     "RES_DNSRCH", RES_DNSRCH,
-    "RES_INSECURE1", RES_INSECURE1,
-    "RES_INSECURE2", RES_INSECURE2,
+/*    "RES_INSECURE1", RES_INSECURE1,
+    "RES_INSECURE2", RES_INSECURE2, */
     "RES_NOALIASES", RES_NOALIASES,
-    "RES_USE_INET6", RES_USE_INET6,
+/*    "RES_USE_INET6", RES_USE_INET6, */
 #ifdef RES_ROTATE
     "RES_ROTATE", RES_ROTATE,
 #endif
