diff --git a/configure.py b/configure.py
index dab5dc9..5a87d68 100755
--- a/configure.py
+++ b/configure.py
@@ -207,6 +207,7 @@ for target in targets:
     # The rule for building a .bc file for the specified architecture using clang.
     clang_bc_flags = "-target %s -I`dirname $in` %s " \
                      "-fno-builtin " \
+                     "-nostdlib " \
                      "-D__CLC_INTERNAL " \
                      "-emit-llvm" % (target, clang_cl_includes)
     if device['gpu'] != '':
