diff -aurN jove4.16.0.73/iproc.c jove4.16.0.73.new/iproc.c
--- jove4.16.0.73/iproc.c	2008-05-21 04:35:18.000000000 +0200
+++ jove4.16.0.73.new/iproc.c	2019-08-03 15:02:30.745170109 +0200
@@ -7,7 +7,7 @@
 
 #include "jove.h"
 
-#ifdef IPROCS	/* the body is the rest of this file */
+/*#ifdef IPROCS	/* the body is the rest of this file */
 
 #include <signal.h>
 
@@ -538,12 +538,12 @@
 #  include <stdlib.h>	/* for grantpt and unlockpt, at least in Solaris 2.3 */
 #  if _XOPEN_SOURCE >= 500
     /* Linux/glibc no longer pretends to support STREAMS (XSR) (2008) */
-#   if _XOPEN_STREAMS != -1
+/*#   if _XOPEN_STREAMS != -1
 #    include <stropts.h>
 #   endif
 #  else
 #   include <sys/stropts.h>
-#  endif
+#  endif*/
   extern char	*ptsname proto((int /*filedes*/));	/* get name of slave */
 # endif
 
