diff -aurN jove4.16.0.73/Makefile jove4.16.0.73-mod/Makefile
--- jove4.16.0.73/Makefile	2019-09-03 05:25:41.332128938 +0000
+++ jove4.16.0.73-mod/Makefile	2019-09-03 05:25:05.492127978 +0000
@@ -138,7 +138,7 @@
 # Add -DUSE_EXIT if you're profiling or using purify (this causes Jove
 # to exit using exit(), instead of _exit()).
 
-SYSDEFS = -DSYSVR4 -D_XOPEN_SOURCE=500
+SYSDEFS = -DSYSVR4 -D_XOPEN_SOURCE=500 -D_XOPEN_STREAMS=-1
 
 # Select optimization level (flags passed to compiling and linking steps).
 # On most systems: -g for debugging, -O for optimization.
