--- mesa-23.0.2/src/gallium/frontends/rusticl/meson.build.orig      2023-04-07 18:22:42.000000000 +0200
+++ mesa-23.0.2/src/gallium/frontends/rusticl/meson.build   2023-04-13 12:36:18.515818599 +0200
@@ -92,7 +92,6 @@ rusticl_gen_args = [
 rusticl_bindgen_args = [
   '--no-convert-floats',
   '--use-array-pointers-in-arguments',
-  '--size_t-is-usize',
   '--default-enum-style', 'rust',
   '--with-derive-partialeq',
   '--with-derive-eq',
