diff -aurN cryfs-0.10.3/src/cpp-utils/thread/LeftRight.h cryfs-0.10.3-mod/src/cpp-utils/thread/LeftRight.h
--- cryfs-0.10.3/src/cpp-utils/thread/LeftRight.h	2021-04-02 19:08:08.000000000 +0200
+++ cryfs-0.10.3-mod/src/cpp-utils/thread/LeftRight.h	2023-05-01 18:11:54.775016196 +0200
@@ -4,6 +4,7 @@
 #include <thread>
 #include <cpp-utils/macros.h>
 #include <array>
+#include <stdexcept>
 
 namespace cpputils {
