diff -aurN xine-lib-1.2.12/include/xine.h xine-lib-1.2.12-mod/include/xine.h
--- xine-lib-1.2.12/include/xine.h	2022-03-06 17:31:23.000000000 +0100
+++ xine-lib-1.2.12-mod/include/xine.h	2022-09-01 13:18:28.330330137 +0200
@@ -118,7 +118,7 @@
 /* allow the setting of some flags before xine_init
  * FIXME-ABI: this is currently GLOBAL
  */
-void xine_set_flags (xine_t *, int) XINE_PROTECTED XINE_WEAK;
+void xine_set_flags (xine_t *, int)  XINE_WEAK;
 #define XINE_FLAG_NO_WRITE_CACHE		1
 
 /*
