diff -aurN efivar-20201015_cff88dd/src/gpt.c efivar-20201015_cff88dd-mod/src/gpt.c
--- efivar-20201015_cff88dd/src/gpt.c	2021-03-13 20:10:54.000000000 +0100
+++ efivar-20201015_cff88dd-mod/src/gpt.c	2022-08-11 09:44:52.178874254 +0200
@@ -17,6 +17,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mount.h>
 #include <sys/param.h>
 #include <sys/stat.h>
 #include <sys/utsname.h>
diff -aurN efivar-20201015_cff88dd/src/linux.c efivar-20201015_cff88dd-mod/src/linux.c
--- efivar-20201015_cff88dd/src/linux.c	2021-03-13 20:10:54.000000000 +0100
+++ efivar-20201015_cff88dd-mod/src/linux.c	2022-08-11 09:45:39.059089300 +0200
@@ -20,6 +20,7 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <sys/ioctl.h>
+#include <sys/mount.h>
 #include <sys/socket.h>
 #include <sys/sysmacros.h>
 #include <sys/types.h>
diff -aurN efivar-20201015_cff88dd/src/util.h efivar-20201015_cff88dd-mod/src/util.h
--- efivar-20201015_cff88dd/src/util.h	2021-03-13 20:10:54.000000000 +0100
+++ efivar-20201015_cff88dd-mod/src/util.h	2022-08-11 09:43:39.980097728 +0200
@@ -21,7 +21,6 @@
 #include <stdio.h>
 #include <string.h>
 #include <sys/ioctl.h>
-#include <sys/mount.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <tgmath.h>
