--- libindicator-0.5.0.orig/libindicator/indicator-service.c
+++ libindicator-0.5.0/libindicator/indicator-service.c
@@ -100,9 +100,9 @@ static void bus_method_call (GDBusConnec
 static GDBusNodeInfo *            node_info = NULL;
 static GDBusInterfaceInfo *       interface_info = NULL;
 static GDBusInterfaceVTable       interface_table = {
-	method_call:	bus_method_call,
-	get_property:	NULL, /* No properties */
-	set_property:	NULL  /* No properties */
+	.method_call = bus_method_call,
+	.get_property = NULL, /* No properties */
+	.set_property = NULL  /* No properties */
 };
 
 /* THE define */
