diff -aurN libindicator-12.10.1/libindicator/Makefile.am libindicator-12.10.1-mod/libindicator/Makefile.am
--- libindicator-12.10.1/libindicator/Makefile.am	2012-09-19 17:38:05.113919565 +0200
+++ libindicator-12.10.1-mod/libindicator/Makefile.am	2019-12-24 12:49:00.616622404 +0100
@@ -57,7 +57,7 @@
 	$(LIBINDICATOR_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall -Wno-deprecated-declarations
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
diff -aurN libindicator-12.10.1/tools/Makefile.am libindicator-12.10.1-mod/tools/Makefile.am
--- libindicator-12.10.1/tools/Makefile.am	2012-09-19 17:38:04.933829559 +0200
+++ libindicator-12.10.1-mod/tools/Makefile.am	2019-12-24 12:49:56.087444990 +0100
@@ -17,7 +17,7 @@
 	indicator-loader.c
 
 indicator_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall -Wno-deprecated-declarations \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
