diff -aurN syslinux-4.07/extlinux/main.c syslinux-4.07-mod/extlinux/main.c
--- syslinux-4.07/extlinux/main.c	2013-07-25 09:28:59.000000000 -0000
+++ syslinux-4.07-mod/extlinux/main.c	2018-08-08 17:39:35.809144894 -0000
@@ -36,6 +36,7 @@
 #include <getopt.h>
 #include <sysexits.h>
 #include <sys/ioctl.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/mount.h>
