diff -ruBN /tmp/aaa_base/aaa_base.SlackBuild /source/a/aaa_base/aaa_base.SlackBuild
--- /tmp/aaa_base/aaa_base.SlackBuild	2018-06-24 18:43:57.000000000 +0000
+++ /source/a/aaa_base/aaa_base.SlackBuild	2021-01-02 12:18:16.000000000 +0000
@@ -80,7 +80,7 @@
 fi
 
 # Add lib64 directories on x86_64:
-if [ "$ARCH" = "x86_64" ]; then
+if [ "$LIBDIRSUFFIX" = "64" ]; then
   mkdir -p $PKG/lib64 $PKG/usr/lib64 $PKG/usr/local/lib64
 fi
 
