diff -aurN bpe-2.01.00/backup.c bpe-2.01.00-mod/backup.c
--- bpe-2.01.00/backup.c	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/backup.c	2020-10-30 17:01:54.667060579 -0000
@@ -65,7 +65,7 @@
 	if ((fdb = open(bfile, O_WRONLY | O_CREAT | O_TRUNC,
 		   (int) (Sbuf.st_mode & 0777))) == -1)
 		{
-		stl_msg(sys_errlist[errno], bfile, BUC1ERR, 1);
+		stl_msg(strerrorname_np[errno], bfile, BUC1ERR, 1);
 		(void) umask(cmask);
 		Fil[optind].bak_status = errbak;
 		return;
@@ -73,7 +73,7 @@
 
 	if (lseek(Crnt.fd, 0L, 0) == -1)
 		{
-		stl_msg(sys_errlist[errno], (char *) NULL, LSERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LSERR, 1);
 		(void) umask(cmask);
 		(void) close(fdb);
 		Fil[optind].bak_status = errbak;
@@ -86,7 +86,7 @@
 			break;
 
 	if (c < 0 || d < 0)
-		stl_msg(sys_errlist[errno], (char *) NULL,
+		stl_msg(strerrorname_np[errno], (char *) NULL,
 		      (c < 0) ? RDERR : WRERR, 1);
 
 	(void) close(fdb);
@@ -251,7 +251,7 @@
 		   O_WRONLY | O_APPEND : O_WRONLY | O_CREAT | O_TRUNC,
 		   (~cmask) & 0666)) == -1)
 		{
-		stl_msg(sys_errlist[errno], filename, (trunc_flag == off) ?
+		stl_msg(strerrorname_np[errno], filename, (trunc_flag == off) ?
 		       FAERR : FCERR, 1);
 		(void) umask(cmask);
 		return;
@@ -259,7 +259,7 @@
 
 	if (lseek(Crnt.fd, start, 0) == -1)
 		{
-		stl_msg(sys_errlist[errno], (char *) NULL, LSERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LSERR, 1);
 		(void) umask(cmask);
 		(void) close(fdf);
 		return;
@@ -278,7 +278,7 @@
 
 	if (d < 0 || e < 0)
 		/* read/write-error message */
-		stl_msg(sys_errlist[errno], (char *) NULL,
+		stl_msg(strerrorname_np[errno], (char *) NULL,
 		      (d < 0) ? RDERR : WRERR, 1);
 
 	else
diff -aurN bpe-2.01.00/bpe.h bpe-2.01.00-mod/bpe.h
--- bpe-2.01.00/bpe.h	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/bpe.h	2020-10-30 16:57:19.236069305 -0000
@@ -64,7 +64,7 @@
 
 #ifdef NEED_SYS_ERRLIST
    /* extern int errno; */
-   extern char *sys_errlist[];
+   extern char *strerrorname_np[];
 #endif
 
 #if (SIZEOF_OFF_T > SIZEOF_LONG) 
diff -aurN bpe-2.01.00/buffer.c bpe-2.01.00-mod/buffer.c
--- bpe-2.01.00/buffer.c	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/buffer.c	2020-10-30 17:02:36.880292780 -0000
@@ -37,15 +37,15 @@
 		dispos();
 
 	if (lseek(Crnt.fd, Fil[optind].fpos, 0) == -1)
-		stl_msg(sys_errlist[errno], (char *) NULL, LSERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LSERR, 1);
 
 	else
 /*		if ((Crnt.bytes = read(Crnt.fd, (char *) Buf.pagebuf,
 		     Ute.edit_chars)) == -1)
-			stl_msg(sys_errlist[errno], (char *) NULL, RDERR, 1); */
+			stl_msg(strerrorname_np[errno], (char *) NULL, RDERR, 1); */
 		if ((Crnt.bytes = read(Crnt.fd, (unsigned char *) Buf.pagebuf,
 		     Ute.edit_chars)) == -1)
-			stl_msg(sys_errlist[errno], (char *) NULL, RDERR, 1);
+			stl_msg(strerrorname_np[errno], (char *) NULL, RDERR, 1);
 
 	return(Crnt.bytes);
 
@@ -167,14 +167,14 @@
 
 	if (lseek(Crnt.fd, Fil[optind].fpos, 0) == -1)
 		{
-		stl_msg(sys_errlist[errno], (char *) NULL, LSERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LSERR, 1);
 		return(-1);
 		}
 
 	if (write(Crnt.fd, (char *) Buf.pagebuf,
 	    (unsigned) Crnt.bytes) != Crnt.bytes)
 		{
-		stl_msg(sys_errlist[errno], (char *) NULL, WRERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, WRERR, 1);
 		return(-1);
 		}
 
@@ -215,7 +215,7 @@
 	   (unsigned) sizeof (struct files))) == NULL)
 		{
 		(void) fprintf(stderr,"%s: cannot allocate memory: %s\n",
-		    Ute.bpe_name, sys_errlist[errno]);
+		    Ute.bpe_name, strerrorname_np[errno]);
 		(void) fflush(stderr);
 		return(1);
 		}
@@ -244,7 +244,7 @@
 	    (unsigned char *) malloc((unsigned) Ute.edit_chars)) == NULL)
 		{
 		(void) sprintf(Ute.errmsg, "cannot allocate buffer memory: %s",
-		    sys_errlist[errno]);
+		    strerrorname_np[errno]);
 		return(1);
 		}
 
@@ -277,7 +277,7 @@
 	    (strlen(filename) + 1))) == NULL)
 		{
 		/* cannot allocate memory message */
-		stl_msg(sys_errlist[errno], (char *) NULL, MEMERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, MEMERR, 1);
 		return(1);
 		}
 
diff -aurN bpe-2.01.00/file.c bpe-2.01.00-mod/file.c
--- bpe-2.01.00/file.c	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/file.c	2020-10-30 17:03:09.194704994 -0000
@@ -205,7 +205,7 @@
 			stl_msg((char *) NULL, last_file, NOSFILE, 1);
 
 		else
-			stl_msg(sys_errlist[errno], last_file, ACCERR, 1);
+			stl_msg(strerrorname_np[errno], last_file, ACCERR, 1);
 
 		return(1);
 		}
@@ -491,7 +491,7 @@
 		{
 		Fil[optind].ftype = 0;
 		dispheader(1, filename);
-		stl_msg(sys_errlist[errno], filename, OPERR, 1);
+		stl_msg(strerrorname_np[errno], filename, OPERR, 1);
 		return(1);
 		}
 
@@ -550,7 +550,7 @@
 	if (close(Crnt.fd) == -1)
 		{
 		(void) sprintf(Ute.errmsg, "cannot close: \"%s\" - %s",
-			       filename, sys_errlist[errno]);
+			       filename, strerrorname_np[errno]);
 		return(1);
 		}
 
@@ -579,7 +579,7 @@
 
 	if (stat(filename, &Sbuf))
 		{
-		stl_msg(sys_errlist[errno], filename, STERR, 1);
+		stl_msg(strerrorname_np[errno], filename, STERR, 1);
 		Fil[optind].flength = -1L;
 		return(-1);
 		}
@@ -630,7 +630,7 @@
 
 	if (stat(filename, &tsbuf))
 		{
-		stl_msg(sys_errlist[errno], filename, STERR, 1);
+		stl_msg(strerrorname_np[errno], filename, STERR, 1);
 		return(1);
 		}
 
@@ -649,7 +649,7 @@
 	tbuf.actime = tsbuf.st_atime;
 	if (utime(filename, &tbuf))
 		{
-		stl_msg(sys_errlist[errno], filename, TIMERR, 1);
+		stl_msg(strerrorname_np[errno], filename, TIMERR, 1);
 		return(1);
 		}
 
diff -aurN bpe-2.01.00/lock.c bpe-2.01.00-mod/lock.c
--- bpe-2.01.00/lock.c	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/lock.c	2020-10-30 17:03:33.998253812 -0000
@@ -36,7 +36,7 @@
 			if (unlink(lockfile))
 				{
 				Crnt.lock_flag = errnolock;
-				stl_msg(sys_errlist[errno],
+				stl_msg(strerrorname_np[errno],
 				      (char *) NULL, LFDERR, 1);
 				}
 			}
@@ -114,7 +114,7 @@
 		else
 			{
 			/* Cannot create lockfile */
-			stl_msg(sys_errlist[errno], (char *) NULL, LFCERR, 1);
+			stl_msg(strerrorname_np[errno], (char *) NULL, LFCERR, 1);
 			Crnt.lock_flag = errnolock;
 			}
 
@@ -128,7 +128,7 @@
 	    sizeof(struct lock_id))
 		{
 		/* Cannot write to lockfile */
-		stl_msg(sys_errlist[errno], (char *) NULL, LFWERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LFWERR, 1);
 		(void) close(fdl);
 		(void) umask(cmask);
 		Crnt.lock_flag = errlock;
@@ -160,7 +160,7 @@
 		else
 			{
 			/* lockfile error */
-			stl_msg(sys_errlist[errno], (char *) NULL, LFERR, 1);
+			stl_msg(strerrorname_np[errno], (char *) NULL, LFERR, 1);
 			Crnt.lock_flag = errlock;
 			return;
 			}
@@ -170,7 +170,7 @@
 	    sizeof(struct lock_id))
 		{
 		/* Cannot read lockfile */
-		stl_msg(sys_errlist[errno], (char *) NULL, LFRERR, 1);
+		stl_msg(strerrorname_np[errno], (char *) NULL, LFRERR, 1);
 		(void) close(fdl);
 		Crnt.lock_flag = errlock;
 		return;
@@ -189,7 +189,7 @@
 
 	if (Crnt.lock_flag == lock || Crnt.lock_flag == errlock)
 		if (unlink(lockfile) && (status_code))
-			stl_msg(sys_errlist[errno], (char *) NULL, LFDERR, 1);
+			stl_msg(strerrorname_np[errno], (char *) NULL, LFDERR, 1);
 
 }
 
diff -aurN bpe-2.01.00/main.c bpe-2.01.00-mod/main.c
--- bpe-2.01.00/main.c	2000-03-19 23:46:30.000000000 -0000
+++ bpe-2.01.00-mod/main.c	2020-10-30 17:03:55.415864210 -0000
@@ -226,7 +226,7 @@
 	if (Ute.numfiles == 1 && access(argv[optind], 4))
 		{
 		(void) fprintf(stderr, "%s: cannot access: \"%s\" - %s\n",
-		    Ute.bpe_name, argv[optind], sys_errlist[errno]);
+		    Ute.bpe_name, argv[optind], strerrorname_np[errno]);
 		(void) fflush(stderr);
 		return(EXIT_FAILURE);
 		}
@@ -452,7 +452,7 @@
 	if ((ttyflag = setup_tty()) != 0)
 		{
 		(void) fprintf(stderr, "%s: ioctl error - %s\n",
-		    Ute.bpe_name, sys_errlist[ttyflag]);
+		    Ute.bpe_name, strerrorname_np[ttyflag]);
 		(void) fflush(stderr);
 		return(EXIT_FAILURE);
 		}
