diff -aurN udisks-1.0.5/src/daemon.c udisks-1.0.5-mod/src/daemon.c
--- udisks-1.0.5/src/daemon.c	2014-03-10 09:57:48.000000000 +0100
+++ udisks-1.0.5-mod/src/daemon.c	2018-08-06 08:17:58.883026627 +0200
@@ -45,6 +45,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/utsname.h>
+#include <sys/sysmacros.h>
 #include <net/if_arp.h>
 #include <fcntl.h>
 #include <signal.h>
diff -aurN udisks-1.0.5/src/device.c udisks-1.0.5-mod/src/device.c
--- udisks-1.0.5/src/device.c	2014-03-10 09:57:48.000000000 +0100
+++ udisks-1.0.5-mod/src/device.c	2018-08-06 08:20:25.129026069 +0200
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/resource.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
diff -aurN udisks-1.0.5/src/mount-monitor.c udisks-1.0.5-mod/src/mount-monitor.c
--- udisks-1.0.5/src/mount-monitor.c	2014-03-10 09:58:42.000000000 +0100
+++ udisks-1.0.5-mod/src/mount-monitor.c	2018-08-06 08:21:11.428025892 +0200
@@ -28,7 +28,9 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
+
 #include <mntent.h>
 
 #include <glib.h>
diff -aurN udisks-1.0.5/tools/udisks.c udisks-1.0.5-mod/tools/udisks.c
--- udisks-1.0.5/tools/udisks.c	2014-03-10 09:57:48.000000000 +0100
+++ udisks-1.0.5-mod/tools/udisks.c	2018-08-06 08:21:57.700025716 +0200
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
diff -aurN udisks-1.0.5/tools/umount-udisks.c udisks-1.0.5-mod/tools/umount-udisks.c
--- udisks-1.0.5/tools/umount-udisks.c	2014-03-10 09:57:48.000000000 +0100
+++ udisks-1.0.5-mod/tools/umount-udisks.c	2018-08-06 08:22:15.498025648 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
