diff -aurN kjots-20211003_4109f07/src/kjotsmodel.cpp kjots-20211003_4109f07-mod/src/kjotsmodel.cpp
--- kjots-20211003_4109f07/src/kjotsmodel.cpp	2021-10-10 23:33:02.000000000 +0200
+++ kjots-20211003_4109f07-mod/src/kjotsmodel.cpp	2022-04-23 19:27:54.826344390 +0200
@@ -21,7 +21,11 @@
 #include <AkonadiCore/ChangeRecorder>
 #include <AkonadiCore/EntityDisplayAttribute>
 #endif
+#if AKONADI_VERSION >= QT_VERSION_CHECK(5, 19, 41)
+#include <Akonadi/NoteUtils>
+#else
 #include <Akonadi/Notes/NoteUtils>
+#endif
 
 #include <KMime/Message>
 #include <KPIMTextEdit/TextUtils>
diff -aurN kjots-20211003_4109f07/src/kjotswidget.cpp kjots-20211003_4109f07-mod/src/kjotswidget.cpp
--- kjots-20211003_4109f07/src/kjotswidget.cpp	2021-10-10 23:33:02.000000000 +0200
+++ kjots-20211003_4109f07-mod/src/kjotswidget.cpp	2022-04-23 19:28:15.128583076 +0200
@@ -34,8 +34,12 @@
 #include <QDebug>
 
 // Akonadi
-#include <Akonadi/Notes/NoteUtils>
 #include <akonadi_version.h>
+#if AKONADI_VERSION >= QT_VERSION_CHECK(5, 19, 41)
+#include <Akonadi/NoteUtils>
+#else
+#include <Akonadi/Notes/NoteUtils>
+#endif
 #if AKONADI_VERSION >= QT_VERSION_CHECK(5, 18, 41)
 #include <Akonadi/AttributeFactory>
 #include <Akonadi/CollectionCreateJob>
diff -aurN kjots-20211003_4109f07/src/noteshared/notecreatorandselector.cpp kjots-20211003_4109f07-mod/src/noteshared/notecreatorandselector.cpp
--- kjots-20211003_4109f07/src/noteshared/notecreatorandselector.cpp	2021-10-10 23:33:02.000000000 +0200
+++ kjots-20211003_4109f07-mod/src/noteshared/notecreatorandselector.cpp	2022-04-23 19:28:38.803861428 +0200
@@ -24,7 +24,11 @@
 #include <AkonadiCore/EntityTreeModel>
 #include <AkonadiCore/ItemCreateJob>
 #endif
+#if AKONADI_VERSION >= QT_VERSION_CHECK(5, 19, 41)
+#include <Akonadi/NoteUtils>
+#else
 #include <Akonadi/Notes/NoteUtils>
+#endif
 
 #include "noteshared_debug.h"
 
diff -aurN kjots-20211003_4109f07/src/noteshared/standardnoteactionmanager.cpp kjots-20211003_4109f07-mod/src/noteshared/standardnoteactionmanager.cpp
--- kjots-20211003_4109f07/src/noteshared/standardnoteactionmanager.cpp	2021-10-10 23:33:02.000000000 +0200
+++ kjots-20211003_4109f07-mod/src/noteshared/standardnoteactionmanager.cpp	2022-04-23 19:28:56.268066765 +0200
@@ -25,7 +25,11 @@
 #include <AkonadiCore/CollectionModifyJob>
 #include <AkonadiCore/ItemModifyJob>
 #endif
+#if AKONADI_VERSION >= QT_VERSION_CHECK(5, 19, 41)
+#include <Akonadi/NoteUtils>
+#else
 #include <Akonadi/Notes/NoteUtils>
+#endif
 
 #include <KXmlGui/KActionCollection>
 #include <KLocalizedString>
