diff -aurN clucene-src-2.3.3.4/src/core/CLucene/document/DateTools.cpp clucene-src-2.3.3.4-mod/src/core/CLucene/document/DateTools.cpp
--- clucene-src-2.3.3.4/src/core/CLucene/document/DateTools.cpp	2011-03-17 01:21:07.000000000 +0100
+++ clucene-src-2.3.3.4-mod/src/core/CLucene/document/DateTools.cpp	2022-10-06 20:29:47.465795041 +0200
@@ -8,6 +8,7 @@
 
 #include "DateTools.h"
 #include "CLucene/util/Misc.h"
+#include <ctime>
 
 CL_NS_USE(util)
 CL_NS_DEF(document)
