diff -aurN parted-3.2/libparted/architecture.h parted-3.2-mod/libparted/architecture.h
--- parted-3.2/libparted/architecture.h	2014-06-15 20:17:43.000000000 +0200
+++ parted-3.2-mod/libparted/architecture.h	2018-08-05 19:08:53.226207236 +0200
@@ -24,6 +24,7 @@
 #define _LIBPARTED_ARCH_H_INCLUDED
 
 #include <parted/disk.h>
+#include <sys/sysmacros.h>
 
 struct _PedArchitecture {
 	PedDiskArchOps*		disk_ops;
