diff -aurN xorg-server-1.20.13/hw/xfree86/ramdac/meson.build xorg-server-21.1.0/hw/xfree86/ramdac/meson.build
--- xorg-server-1.20.13/hw/xfree86/ramdac/meson.build	2021-07-29 20:48:14.000000000 +0200
+++ xorg-server-21.1.0/hw/xfree86/ramdac/meson.build	2021-10-27 12:47:08.000000000 +0200
@@ -1,11 +1,6 @@
 srcs_xorg_ramdac = [
-    'xf86RamDac.c',
-    'xf86RamDacCmap.c',
     'xf86CursorRD.c',
     'xf86HWCurs.c',
-    'IBM.c',
-    'BT.c',
-    'TI.c',
 ]
 
 xorg_ramdac = static_library('xorg_ramdac',
@@ -17,11 +12,7 @@
 
 install_data(
     [
-        'BT.h',
-        'IBM.h',
-        'TI.h',
         'xf86Cursor.h',
-        'xf86RamDac.h',
     ],
     install_dir: xorgsdkdir,
 )
