--- a/SWIG/_rsa.i 2019-05-31 16:59:57.000000000 +0200
+++ b/SWIG/_rsa.i       2023-03-18 16:54:32.729035629 +0100
@@ -23,7 +23,9 @@
 
 %constant int no_padding        = RSA_NO_PADDING;
 %constant int pkcs1_padding     = RSA_PKCS1_PADDING;
+#if OPENSSL_VERSION_MAJOR < 3
 %constant int sslv23_padding    = RSA_SSLV23_PADDING;
+#endif
 %constant int pkcs1_oaep_padding = RSA_PKCS1_OAEP_PADDING;
 
 %constant int NID_sha1 = NID_sha1;
