diff -aurN libappindicator-12.10.0/src/Makefile.am libappindicator-12.10.0-mod/src/Makefile.am
--- libappindicator-12.10.0/src/Makefile.am	2012-07-11 19:28:34.415113869 +0200
+++ libappindicator-12.10.0-mod/src/Makefile.am	2019-12-24 13:38:57.015051155 +0100
@@ -76,7 +76,7 @@
 libappindicator_la_CFLAGS = \
 	$(LIBRARY_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
-	-Wall -Werror \
+	-Wall -Wno-deprecated-declarations \
 	-DG_LOG_DOMAIN=\"libappindicator\"
 
 libappindicator_la_LIBADD = \
