diff -aurN syslinux-4.07/extlinux/main.c syslinux-4.07-mod/extlinux/main.c
--- syslinux-4.07/extlinux/main.c	2013-07-25 11:28:59.000000000 +0200
+++ syslinux-4.07-mod/extlinux/main.c	2022-08-14 11:24:10.831322463 +0200
@@ -38,7 +38,6 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-#include <sys/mount.h>
 #include <sys/vfs.h>
 
 #include "linuxioctl.h"
diff -aurN syslinux-4.07/libinstaller/syslxcom.c syslinux-4.07-mod/libinstaller/syslxcom.c
--- syslinux-4.07/libinstaller/syslxcom.c	2013-07-25 11:28:59.000000000 +0200
+++ syslinux-4.07-mod/libinstaller/syslxcom.c	2022-08-11 20:02:16.904180709 +0200
@@ -28,7 +28,6 @@
 #include <errno.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-#include <sys/mount.h>
 #include <sys/vfs.h>
 
 #include "linuxioctl.h"
diff -aurN syslinux-4.07/linux/syslinux.c syslinux-4.07-mod/linux/syslinux.c
--- syslinux-4.07/linux/syslinux.c	2013-07-25 11:28:59.000000000 +0200
+++ syslinux-4.07-mod/linux/syslinux.c	2022-08-14 11:20:51.763591684 +0200
@@ -45,7 +45,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <sys/mount.h>
 
 #include "linuxioctl.h"
 
