diff -aurN xf86-video-vmware-13.3.0/vmwgfx/vmwgfx_tex_video.c xf86-video-vmware-13.3.0-mod/vmwgfx/vmwgfx_tex_video.c
--- xf86-video-vmware-13.3.0/vmwgfx/vmwgfx_tex_video.c	2018-05-17 09:42:24.000000000 +0200
+++ xf86-video-vmware-13.3.0-mod/vmwgfx/vmwgfx_tex_video.c	2022-08-03 18:58:09.951798799 +0200
@@ -237,7 +237,7 @@
 
        for (i=0; i<3; ++i) {
 	   for (j=0; j<2; ++j) {
-	       if (priv->yuv[i]) {
+	       if (priv->yuv[j][i]) {
 		   xa_surface_destroy(priv->yuv[j][i]);
 		   priv->yuv[j][i] = NULL;
 	       }
